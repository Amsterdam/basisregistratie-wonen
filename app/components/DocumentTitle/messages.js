import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  default_title: {
    id: `${scope}.registration_plural`,
  },
});
