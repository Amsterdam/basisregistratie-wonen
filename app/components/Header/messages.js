/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
  },
  log_in: {
    id: `${scope}.log_in`,
  },
  log_in_adw: {
    id: `${scope}.log_in_adw`,
  },
  log_out: {
    id: `${scope}.log_out`,
  },
});
