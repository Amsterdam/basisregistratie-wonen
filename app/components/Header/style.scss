button {
  appearance: none;
  background: none;
  border: 0;
}
