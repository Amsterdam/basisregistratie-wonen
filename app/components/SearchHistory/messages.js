/*
 * SearchHistory Messages
 *
 * This contains all the text for the SearchHistory component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_history_header: {
    id: `${scope}.search_history_header`,
    defaultMessage: 'Recently search for',
  },
});
