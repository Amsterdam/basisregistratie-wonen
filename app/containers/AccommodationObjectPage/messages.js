/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
