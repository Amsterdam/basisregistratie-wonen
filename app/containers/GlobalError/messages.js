/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
