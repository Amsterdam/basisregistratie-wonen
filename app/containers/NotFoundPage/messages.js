/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  document_title: {
    id: `${scope}.registration_plural`,
  },
  header: {
    id: `${scope}.page_not_found`,
  },
});
