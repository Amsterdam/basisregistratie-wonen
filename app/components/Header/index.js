import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { intlShape } from 'react-intl';

import SearchContainer from 'containers/Search';
import CONFIGURATION from 'shared/services/configuration/configuration';
import loginIcon from '@datapunt/asc-assets/lib/Icons/Login.svg';
import logoutIcon from '@datapunt/asc-assets/lib/Icons/Logout.svg';
import { Header as HeaderComponent, Icon } from '@datapunt/asc-ui';
import messages from './messages';

const StyledHeader = styled(HeaderComponent)`
  a {
    line-height: normal;
  }
`;

const HeaderWrapper = styled.div`
  position: relative;
`;

const StyledNav = styled.nav`
  position: absolute;
  top: calc(78px / 2 - 27px / 2);
  right: 0;
  z-index: 101;

  button {
    appearance: none;
    background: none;
    border: 0;
    cursor: pointer;

    & > * {
      vertical-align: middle;
      margin-right: 5px;
    }

    &:hover {
      color: #ec0000;
      text-decoration: underline;
    }
  }
`;

const Header = ({ isAuthenticated, intl, onLoginLogoutButtonClick }) => (
  <HeaderWrapper>
    <StyledHeader title="Basisregistratie Wonen" homeLink={CONFIGURATION.ROOT} tall fullWidth={false}>
      <SearchContainer />
    </StyledHeader>

    <StyledNav>
      <ul className="links horizontal">
        {!isAuthenticated && (
          <>
            <li>
              <button type="button" onClick={event => onLoginLogoutButtonClick(event, 'datapunt')}>
                <Icon iconUrl={`url('${loginIcon}');`} padding={0} inline />
                {intl.formatMessage(messages.log_in)}
              </button>
            </li>
            <li>
              <button type="button" onClick={event => onLoginLogoutButtonClick(event, 'grip')}>
                <Icon iconUrl={`url('${loginIcon}');`} padding={0} inline />
                {intl.formatMessage(messages.log_in_adw)}
              </button>
            </li>
          </>
        )}
        {isAuthenticated ? (
          <li>
            <button type="button" onClick={onLoginLogoutButtonClick}>
              <Icon iconUrl={`url('${logoutIcon}');`} padding={0} inline />
              {intl.formatMessage(messages.log_out)}
            </button>
          </li>
        ) : (
          ''
        )}
      </ul>
    </StyledNav>
  </HeaderWrapper>
);

Header.defaultProps = {
  isAuthenticated: false,
  onLoginLogoutButtonClick: undefined,
};

Header.propTypes = {
  isAuthenticated: PropTypes.bool,
  intl: intlShape.isRequired,
  onLoginLogoutButtonClick: PropTypes.func,
};

export default Header;
