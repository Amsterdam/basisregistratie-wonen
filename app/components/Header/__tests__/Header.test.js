import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from '@datapunt/asc-ui';
import 'jest-styled-components';
import Header from '..';
import messages from '../../../translations/nl.json';

const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

describe('Header', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container, rerender } = render(
      <ThemeProvider>
        <Header intl={intl} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();

    rerender(
      <ThemeProvider>
        <Header intl={intl} isAuthenticated />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render log in buttons', () => {
    const { getByText } = render(
      <ThemeProvider>
        <Header intl={intl} />
      </ThemeProvider>,
    );

    expect(getByText('Inloggen')).not.toBeUndefined();
    expect(getByText('Inloggen ADW')).not.toBeUndefined();
  });

  it('should render a log out button', () => {
    const { getByText } = render(
      <ThemeProvider>
        <Header intl={intl} isAuthenticated />
      </ThemeProvider>,
    );

    expect(getByText('Uitloggen')).not.toBeUndefined();
  });
});
