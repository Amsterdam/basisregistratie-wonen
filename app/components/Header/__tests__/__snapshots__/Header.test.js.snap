// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c1 {
  max-width: 1080px;
}

.c0 {
  position: relative;
  z-index: 2;
}

.c2 {
  position: absolute;
  top: 5px;
  right: 20px;
  z-index: 101;
}

.c2 button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: 0;
  cursor: pointer;
}

.c2 button > * {
  vertical-align: middle;
  margin-right: 5px;
}

.c2 button:hover,
.c2 button:focus {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 button:hover svg,
.c2 button:focus svg {
  fill: #ec0000;
}

@media (min-width:1200px) {
  .c0 {
    z-index: 0;
  }
}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .c2 button {
    height: 30px;
  }
}

@media (max-width:720px) {
  .c2 {
    top: 0;
    right: 0;
  }

  .c2 button {
    width: 50px;
    height: 50px;
    padding: 0;
  }

  .c2 .links li {
    margin: 0;
  }

  .c2 .login-adw,
  .c2 span {
    display: none;
  }
}

<div>
  <div
    class="c0"
    data-testid="site-header"
  >
    <div
      class="sc-TOsTZ jhxQdU"
      id="header"
    >
      <header
        class="sc-gzVnrw c1 sc-fAjcbJ eiECdr"
      >
        <h1
          class="sc-kjoXOD jEhmHa"
        >
          <a
            class="sc-bwzfXH gajXgj"
            href="https://localhost:8080/"
          >
            <span
              class="sc-bdVaJa cPtgyk"
            />
          </a>
          <a
            class="sc-gisBJw eoTepN sc-gZMcBi kDLxnL sc-EHOje htmEbB"
            href="https://localhost:8080/"
          >
            Registraties
          </a>
        </h1>
        <nav
          class="sc-cHGsZl dFLCjq"
        >
          <nav
            class="no-print c2"
          >
            <ul
              class="links horizontal"
            >
              <li>
                <button
                  class="login"
                  type="button"
                >
                  <svg
                    focusable="false"
                    viewBox="0 0 36 36"
                    width="20"
                  >
                    <path
                      d="M20 2v4h10v24H20v4h14V2H20z"
                    />
                    <path
                      d="M18.82 8.81L16 11.64 20.36 16H2v4h18.36L16 24.36l2.82 2.83 9.2-9.19-9.2-9.19z"
                    />
                  </svg>
                  <span>
                    Inloggen
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </nav>
        <div
          class="sc-caSCKo cVWIld"
        />
      </header>
    </div>
  </div>
</div>
`;

exports[`Header matches the snapshot 2`] = `
<div>
  <div
    class="Header__HeaderWrapper-sc-15eqjjx-1 eSRqiv"
    data-testid="site-header"
  >
    <div
      class="sc-TOsTZ jhxQdU"
      id="header"
    >
      <header
        class="sc-gzVnrw Header__StyledHeader-sc-15eqjjx-0 NarrX sc-fAjcbJ eiECdr"
      >
        <h1
          class="sc-kjoXOD jEhmHa"
        >
          <a
            class="sc-bwzfXH gajXgj"
            href="https://localhost:8080/"
          >
            <span
              class="sc-bdVaJa cPtgyk"
            />
          </a>
          <a
            class="sc-gisBJw eoTepN sc-gZMcBi kDLxnL sc-EHOje htmEbB"
            href="https://localhost:8080/"
          >
            Registraties
          </a>
        </h1>
        <nav
          class="sc-cHGsZl dFLCjq"
        >
          <nav
            class="no-print Header__StyledNav-sc-15eqjjx-2 aAMtn"
          >
            <ul
              class="links horizontal"
            >
              <li>
                <button
                  type="button"
                >
                  <svg
                    focusable="false"
                    viewBox="0 0 36 36"
                    width="20"
                  >
                    <path
                      d="M16 6V2H2v32h14v-4H6V6h10z"
                    />
                    <path
                      d="M24.82 8.81L22 11.64 26.36 16H8v4h18.36L22 24.36l2.82 2.83 9.2-9.19-9.2-9.19z"
                    />
                  </svg>
                  <span>
                    Uitloggen
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </nav>
        <div
          class="sc-caSCKo cVWIld"
        />
      </header>
    </div>
  </div>
</div>
`;
